### PR TITLE
Show location preview on mobile map view

### LINKED ISSUE #704: Fix list view scrolling on mobile
.On mobile, when you select a location it renders the content scrolled part-way down. Also, the searchbar and filters also exist in a scrollable container and can disappear when they should always be visible. See below for both examples:

Screen Shot 2020-09-25 at 12 37 57 PM

### PR DESCRIPTION
**Pull request by bbovenzi**
Fixes #704, #694, #702 

When clicking on a pin on mobile a preview displays on the bottom of the screen.
Clicking on that preview will expand to show details.
Clicking on the back button goes back to the map.
Also, toggling between list and map view will always show the list or map instead of details.

Other notes: Did a bit of code cleanup to reuse more data and make the code a bit more readable.

Screen Shot 2020-09-25 at 2 26 02 PM

Note: What design changes could we do to not waste so much screen space on smaller mobile devices?
Screen Shot 2020-09-25 at 2 25 54 PM

### CODE REVIEW

**Comment by bbovenzi** [comment]
@pageignited Let me know what you think as well!

File: client/src/components/ResultsContainer.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import React from "react";
+import React, { useEffect, useState } from "react";
```
Comment (bbovenzi) [inline client/src/components/ResultsContainer.js:1]: Reorganized state and effects to better see all of the state at once. Also, removed some redundant state variables

File: client/src/components/ResultsList.js [hunk 2]
```diff
@@ -1,24 +1,19 @@
...
 import { Grid } from "@material-ui/core";
 import { makeStyles } from "@material-ui/core/styles";
-import pantryIcon from "../images/pantryIcon";
-import mealIcon from "../images/mealIcon";
-import splitPantryMealIcon from "../images/splitPantryMealIcon";
-import {
-  MEAL_PROGRAM_CATEGORY_ID,
-  FOOD_PANTRY_CATEGORY_ID,
-} from "../constants/stakeholder";
-import { ORGANIZATION_COLORS, CLOSED_COLOR } from "../constants/map";
```
Comment (bbovenzi) [inline client/src/components/ResultsList.js:15]: Moved StakeholderPreview and Details logic to their own components or a shared helper function.

**Comment by pageignited** [comment]
On what we can do to make the buttons fit. What if we changed the "Pantry and Meals " to text links with a small icon and not the big button. Something like the "list " link you have now.

On the preview pane on let's remove or reduce the size of that square  image icon by 40-50% 
Remove the map pin icon as well under the "mi" text

File: client/src/components/ResultsContainer.js (hunk 1 above)
Comment (entrotech) [reply client/src/components/ResultsContainer.js:1]: good.

File: client/src/components/ResultsList.js (hunk 2 above)
Comment (entrotech) [reply client/src/components/ResultsList.js:15]: Good. We need to do more re-factorign into components where we can.

**Review by entrotech — APPROVED** [review]
Really good work!  I ran your branch on my machine, and it's a big improvement ofver what we had in terms of screen usage.  When you click on a marker on the map, it re-centers the map over the marker. I find this a bit disorienting - what do you think?  There are a number of dead components I want to clean-up, but then our components folder structure is due for a re-organization - I'd be interested in your thoughts on a good folder structure.

**Comment by pageignited** [comment]
yes, I think it should not center over the pinned icon, since you might lose your orientation on the other location in area  a food seeker might want to check out on the map.

**Comment by pageignited** [comment]
On the map view, when you click the pin location, the location tag still shows we need to remove that. Since we have the window at the bottom now, the tag showing on the map is no longer needed.

**Comment by bbovenzi** [comment]
@entrotech Yeah, I think we should make more folders to organize our components better. Probably start with something like `volunteer`, `foodseeker`, and `shared`.

Let me remove the auto-center and clean up some of the UI issues @pageignited found

**Comment by bbovenzi** [comment]
Actually, @pageignited I think the location tag is still necessary since you need to see which one is selected. But I think we could make it cleaner by making the active pin larger and adding an opacity to all others. (Ex: Active pin is 1.25px bigger, all other pins are at 0.9 opacity, and then closed locations are at 0.5 opacity)

**Comment by pageignited** [comment]
@bbovenzi I have in ticket  #699 the maker turning red, and that becomes the active one. Let me know if that can happen or do we need to keep the tag showing?

**Pull request merged**